nanochat/engine.py

```diff
@@ -327,8 +328,11 @@ def generate_batch(self, tokens, num_samples=1, **kwargs):
     import time
     # init compute
     ddp, ddp_rank, ddp_local_rank, ddp_world_size, device = compute_init()
+    device_type = autodetect_device_type()
+    autocast_ctx = torch.amp.autocast(device_type=device_type, dtype=torch.bfloat16) if device_type == "cuda" else nullcontext()
+
     # load the model and tokenizer
-    model, tokenizer, meta = load_model("base", device, phase="eval")
+    model, tokenizer, meta = load_model("sft", device, phase="eval")
```

Comment:
Just noting that this edit is unrelated to the main fix of this PR, though I can see how you'd like to see the output of the SFT model instead of the base one. It doesn't really matter in the scheme of things here though.